src/main/java/frc/robot/scoreStates/ScoreStateContext.java

```diff
@@ -0,0 +1,28 @@
+package frc.robot.scoreStates;
+
+public class ScoreStateContext {
+    private static ScoreState currentState;
+    private static ScoreStateContext INSTANCE;
+
+    public ScoreStateContext getInstance() {
```

Comment:
This class doesn't have a special constructor so I think you can just it statically

Reply:
Actually shouldn't the initial state be passed in the getInstance?

Reply:
I don't think it's necessary, you can call `ScoreStateContext.setCurrentState()` in `robotInit()`, for example.

Reply:
fd842a3

Reply:
States should be saved as class members, otherwise, we'll needlessly create new objects every time the state switches.

Reply:
559704b

Reply:
559704b